lib/takeaway.rb

```diff
@@ -0,0 +1,36 @@
+require 'rubygems'
+require 'twilio-ruby'
+
+class Takeaway
+  attr_reader :order_klass, :order, :order_time
+
+  def initialize(order_klass)
+    @order_klass = order_klass
+    @order = nil
+  end
+
+  def create_order(menu_klass)
+    @order = order_klass.new(menu_klass)
+  end
+
+  def place_order(order, quantity, total, number)
+    fail "Wrong total!" if total != order.calculate_cost(order)
+    @order_time = Time.now + 3600
+    send_sms(number)
+  end
+
+  private
+
+  TWILIO_SMS_NUMBER = '+441922214177'
+  def send_sms(number)
+    account_sid = 'ACf048c7f8c1cffaed9ccf704996f1f604'
+    auth_token = 'b6363d653bfcb190f242096270f12f71'
+    @client = Twilio::REST::Client.new account_sid, auth_token
+    @client.messages.create(
+      from: TWILIO_SMS_NUMBER,
+      to: number,
+      body: "Thank you! Your order was placed and will be delivered before #{order_time}"
```

Comment:
Metrics/LineLength: Line is too long. [89/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)